src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -261,6 +261,18 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 								x = x2;
 							}
 							T t = info.c.newInstance();
+							if (getParser().getCurrentStructure() != null) {
```

Comment:
Just to be extra safe, I think we should check the current sections even if the current structure is null

Reply:
oki shall do c: